tasktiger/redis_semaphore.py

```diff
@@ -0,0 +1,35 @@
+"""Redis Semaphore lock."""
+
+import os
+import time
+
+class Semaphore(object):
+    def __init__(self, redis, name, id, timeout, max=1):
+        self.redis = redis
+        self.name = name
+        self.id = id
+        self.max = max
+        self.timeout = timeout
+        with open(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                               'lua/semaphore.lua')) as f:
+            self._semaphore = self.redis.register_script(f.read())
+
+    def release(self):
+        """Release semaphore."""
+
+        self.redis.zrem(self.name, self.id)
+
+    def acquire(self):
```

Comment:
We could add an optional `def acquire(self, time=time.time):` parameter, pass time explicitly in tests and avoid freezefrog dependency.